include/oneapi/dpl/internal/dynamic_selection_impl/submission_group.h

```diff
@@ -18,31 +18,29 @@ namespace dpl {
 namespace experimental {
   template<typename T>
   class submission_group_t {
-    using submission_container_t = util::concurrent_queue<T>;
+    using submission_container_t = std::shared_ptr<util::concurrent_queue<T>>;
     using submissions_t = std::vector<T>;
     submission_container_t submission_container;
     submissions_t submissions;
```

Comment:
What's the difference of `submission_container` and `submissions`? It seems the latter is not used anywhere.